Library/Homebrew/cmd/update-report.rb

```diff
@@ -630,6 +630,11 @@ def add(reporter, auto_update: false)
 
   def dump(updated_formula_report: true)
     report_all = Homebrew::EnvConfig.update_report_all_formulae?
+    if report_all && !Homebrew::EnvConfig.no_install_from_api?
+      opoo "HOMEBREW_UPDATE_REPORT_ALL_FORMULAE is set but HOMEBREW_NO_INSTALL_FROM_API is not!"
+      puts "This will not report all formula because Homebrew cannot get this data from the API."
```

Comment:
I feel this encourages people to set `HOMEBREW_NO_INSTALL_FROM_API` rather than just unset `HOMEBREW_UPDATE_REPORT_ALL_FORMULAE`.

Reply:
@Bo98 Yup, fair point: deprecated instead.